### ISSUE TITLE
ModuleNotFound error with pylint-django plugin - pylint 3.0.2, pylint_django 2.5.5

### ISSUE THREAD

**Issue by danryu** [pylint-dev/pylint#9283]
### Bug description

```python
Seems to be a regression/repeat of https://github.com/pylint-dev/pylint/issues/7938 with pylint 3.x

Works fine with 2.15.9, 2.17.7.

Fails with ModuleNotFound error with 3.0.2.
```

### Command used

```shell
pylint --load-plugins pylint_django --django-settings-module=myapp.settings myapp/
```

### Pylint output

```python
ModuleNotFoundError: No module named 'somemodule'
```

### Expected behavior

successful pylint test output

### Pylint version

```shell
pylint 3.0.2
```

### OS / Environment

ubuntu 22.04

**Comment by DanielNoord** [comment]
Could you report this to `pylint_django`?
